lib/cn/mbset.c

```diff
@@ -132,39 +76,68 @@ mbset_apply(struct mbset *self, mbset_udata_init_fn *cb, uint *argcp, u64 *argv)
 static merr_t
 mbset_mblk_del(struct mbset *self)
 {
-    merr_t err;
-    uint   i;
-
-    for (i = 0; i < self->mbs_idc; i++) {
-        if (self->mbs_idv[i]) {
-            err = mpool_mblock_delete(self->mbs_mp, self->mbs_idv[i]);
-            if (ev(err))
-                return err;
+    merr_t err_report = 0, err;
+    uint count = 0;
+
+    for (uint i = 0; i < self->mbs_mblkc; i++) {
+        if (self->mbs_mblkv[i].mbid) {
+            err = mpool_mblock_delete(self->mbs_mp, self->mbs_mblkv[i].mbid);
```

Comment:
Here's where we need the mblock ID in the mblock descriptor after it is unmapped.